consumptions/tests/test_views.py

```diff
@@ -2,55 +2,42 @@
 from django.urls import reverse
 import json
 from rest_framework import status
-from rest_framework.test import APIClient, APIRequestFactory
-from .factories import Consumption_Factory
-from consumptions.views import ConsumptionViewSet
+from rest_framework.test import APIClient
+from consumptions.models import Product_Category, Product
 
 
 class TestViews(TestCase):
 
     def setUp(self):
         self.client = APIClient()
-        self.factory = APIRequestFactory()
 
-    def test_index_get(self):
+    def test_index_view(self):
         """Test retrieving the home page"""
         res = self.client.get(reverse('consumptions:index'))
         self.assertEqual(res.status_code, status.HTTP_200_OK)
         self.assertTemplateUsed(res, 'index.html')
 
-    def test_consumption_table_get(self):
+    def test_consumption_table_view(self):
         """Test retrieving a consumptions list"""
         res = self.client.get(reverse('consumptions:consumption_list'))
         self.assertEqual(res.status_code, status.HTTP_200_OK)
         self.assertTemplateUsed(res, 'consumption_list.html')
 
-    def test_consumption_create_get(self):
+    def test_consumption_create_view(self):
         """Test retrieving the create view"""
         res = self.client.get(reverse('consumptions:consumption_create'))
         self.assertEqual(res.status_code, status.HTTP_200_OK)
         self.assertTemplateUsed(res, 'consumption_form.html')
 
-    # def test_consumption_create(self):
-    #     """Test creating consumption"""
-    #     consumption = {
-    #         'product': 'LL001',
-    #         'quantity': 10
-    #     }
-    #     request = self.factory.post('/consumptions-api', consumption, format='json')
-    #     print(request.body)
-    #     view = ConsumptionViewSet
-    #     response = view.as_view(actions={'post': 'create'})(request)
-    #     self.assertEqual(response.status_code, status.HTTP_201_CREATED)
-
-    # def test_create(self):
-    #     data = json.dumps({
-    #         'product': 'LL001',
-    #         'quantity': 10
-    #     })
-    #     response = client.post('/consumptions/create/', data=data, content_type='application/json')
-    #     # Check if you get a 200 back:
-    #     print(response)
-    #     self.assertEqual(response.status_code, )
-    #     # Check to see if Wishbone was created
-    #     self.assertEqual(response.data['results']['product'], 'LL001')
+    def test_create_consumption(self):
+        """Test consumption creating"""
+        prod_cat = Product_Category(code='AA', name='adsl')
+        prod_cat.save()
+        prod = Product(code='AA00', name='aproduct', category=prod_cat)
+        prod.save()
+        data = json.dumps({
+            'product': prod.id,
+            'quantity': 10
+        })
+        response = self.client.post('/consumptions-api/', data=data, content_type='application/json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```

Comment:
I think this is fine for now. The tests look better now but as a next step, we should check that indeed the consumption is created. This assertion alone doesn't guarantee that. It could be that we responded with this status code but the creation failed.